src/libtorch.cc

```diff
@@ -587,8 +590,49 @@ ModelInstanceState::ValidateTypedSequenceControl(
 }
 
 TRITONSERVER_Error*
-ModelInstanceState::ValidateInputs()
+ModelInstanceState::ValidateInputs(const size_t expected_input_cnt)
 {
+  const torch::jit::Method& method = torch_model_->get_method("forward");
+  const auto& schema = method.function().getSchema();
+  const std::vector<c10::Argument>& arguments = schema.arguments();
+
+  // Currently we only support input of type Dict[str, Tensor] when the
+  // model expects a single input. If the model expects more than one input then
+  // they must be all be of type Tensor.
+  //
+  // Ignore the argument at idx 0 if it is of Class type (self param in forward
+  // function)
+  size_t start_idx = 0;
+  if ((arguments.size() > 0) &&
+      (arguments.at(0).type()->kind() == c10::TypeKind::ClassType)) {
+    start_idx = 1;
+  }
+  if ((arguments.size() == (1 + start_idx)) &&
+      (arguments.at(start_idx).type()->kind() == c10::TypeKind::DictType)) {
```

Comment:
Add a block for other c10::Typekinds and throw appropriate error.

Reply:
The error is thrown in the section below where I check if the inputs are of type Tensor

Reply:
Ahh.. I see you have `arguments.size() > start_idx` instead of `arguments.size() > 1 + start_idx` below.

Reply:
Wouldn't it be better to return different errors for multiple Dict(str, Tensor) inputs vs unsupported TypeKind?